core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/LoggingMetricPublisher.java

```diff
@@ -15,27 +15,174 @@
+ * [18e5092e]         │ AvailableConcurrency=0                 │
+ * [18e5092e]         │ LeasedConcurrency=0                    │
+ * [18e5092e]         │ ConcurrencyAcquireDuration=PT0.235851S │
+ * [18e5092e]         │ PendingConcurrencyAcquires=0           │
+ * [18e5092e]         │ MaxConcurrency=50                      │
+ * [18e5092e]         │ HttpClientName=NettyNio                │
+ * [18e5092e]         └────────────────────────────────────────┘
+ * </pre>
+ * Note that the output format may be subject to small changes in future versions and should not be relied upon as a strict public
+ * contract.
  */
 @SdkPublicApi
 public final class LoggingMetricPublisher implements MetricPublisher {
+
+    public enum Format {
+        PLAIN,
+        PRETTY
+    }
+
     private static final Logger LOGGER = Logger.loggerFor(LoggingMetricPublisher.class);
+    private static final Integer PRETTY_INDENT_SIZE = 4;
+
+    private final Level logLevel;
+    private final Format format;
 
-    private LoggingMetricPublisher() {
+    private LoggingMetricPublisher(Level logLevel, Format format) {
+        this.logLevel = Validate.notNull(logLevel, "logLevel");
+        this.format = Validate.notNull(format, "format");
     }
 
+    /**
+     * Create a {@link LoggingMetricPublisher} with the default configuration of {@link Level#INFO} and {@link Format#PLAIN}.
+     */
     public static LoggingMetricPublisher create() {
-        return new LoggingMetricPublisher();
+        return new LoggingMetricPublisher(Level.INFO, Format.PLAIN);
+    }
+
+    /**
+     * Create a {@link LoggingMetricPublisher} with a custom {@link Format} and log {@link Level}.
+     *
+     * @param logLevel the SLF4J log level to log metrics with
+     * @param format   the format to print the metrics with (see class-level documentation for examples)
+     */
+    public static LoggingMetricPublisher create(Level logLevel, Format format) {
+        return new LoggingMetricPublisher(logLevel, format);
     }
 
     @Override
-    public void publish(MetricCollection metricCollection) {
-        LOGGER.info(() -> "Metrics published: " + metricCollection);
+    public void publish(MetricCollection metrics) {
+        if (!LOGGER.isLoggingLevelEnabled(logLevel)) {
+            return;
+        }
+        switch (format) {
+            case PLAIN:
+                LOGGER.log(logLevel, () -> "Metrics published: " + metrics);
+                break;
+            case PRETTY:
+                String guid = Integer.toHexString(metrics.hashCode());
+                logPretty(guid, metrics, 0);
+                break;
+            default:
+                throw new IllegalStateException("Unsupported format: " + format);
+        }
+    }
+
+    private void logPretty(String guid, MetricCollection metrics, int indent) {
+        // Pre-determine metric key-value-pairs so that we can calculate the necessary padding
+        List<String> kvps = new ArrayList<>();
```

Comment:
Nit: the name `kvps` is jarring in the midst of all reader-friendly code. Can't it be `metricValues` or `formattedMetric` or similar?

Reply:
`metricValues` sounds better!